bitcoin/src/string.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Bitcoin string parsing utilities.
+//!
+//! This module provides utility types and traits
+//! to support handling and parsing strings within `rust-bitcoin`.
+
+use crate::prelude::String;
+
+/// Hex parsing error
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[non_exhaustive]
+pub enum FromHexError<E> {
```

Comment:
This error type is missing implementations of `Display` and `std::error::Error` (you can see `blockdata::locktime::absolute::Error` for examples of these). While you are at it, if its not too much bother can you put the error below the trait. I'm on a personal mission to put the most important things first in files :)

Reply:
Yea, that makes sense

Reply:
Damn, I feel quite embarrassed about not catching this one. :D

Reply:
Yeah I thought the same when I saw it, I actually didn't believe that we would have missed it and re-read the whole PR thread :)